lib/web_ui/lib/src/engine/keyboard_binding.dart

```diff
@@ -369,22 +381,33 @@ class KeyboardConverter {
     final String eventKey = event.key!;
 
     final int physicalKey = _getPhysicalCode(event.code!);
-    final bool logicalKeyIsCharacter = !_eventKeyIsKeyname(eventKey);
-    final String? character = logicalKeyIsCharacter ? eventKey : null;
-    final int logicalKey = () {
+    final bool logicalKeyIsCharacter = !_eventKeyIsKeyName(eventKey);
+    final ValueGetter<int> logicalKey = _cached<int>(() {
```

Comment:
This function might not be used: if the event is a key up event, the logical key is simply the pressed logical key, and this function will not be run. In this way, I'm hoping not to run the function body, since it's not a trivial amount of work. And as far as I understand, `late` still always runs the code. (I can add a comment to it.)

Reply:
According to the docs, the `late` keyword makes the field lazy, which means the right-hand-side won't run until you access the field. At least that's my understanding.

https://dart.dev/null-safety/understanding-null-safety#lazy-initialization